DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h

```diff
@@ -203,6 +202,28 @@ class NoiseMap
     return std::ceil((1. + 1. / t) / (relErr * relErr));
   }
 
+  NoiseMap merge(NoiseMap* prev)
+  {
+    int incre = 0;
+    for (size_t i = 0; i < (int)mNoisyPixels.size(); ++i) {
+      for (const auto& prev_np : prev->mNoisyPixels[i]) { // only enters this for loop if the "i" chip exists.
+        bool existsInCurrentMap = false;
+        for (const auto& current_np : mNoisyPixels[i]) {
+          if (prev_np.first == current_np.first) {
+            existsInCurrentMap = true;
+            break;
+          }
+        } // end of for loop on elements of previous noise map
+
+        if (!existsInCurrentMap) {
+          incre++;
+          mNoisyPixels[i][prev_np.first] = prev_np.second;
+        }
```

Comment:
You don't need this loop: since the `mNoisyPixels` is `std::map`, you can do
```
if (mNoisyPixels[i].find(prev_np.first) == mNoisyPixels[i].end()) {
 mNoisyPixels[i][prev_np.first] = prev_np.second;
 incre++;
}
```

Reply:
I think what I am trying to do and the loop you suggested do different things. I am trying to make an intersection of two maps "current_np" and "prev_np"

Reply:
If I got it right, you are looping through all the entries of `mNoisyPixels[i]` to check if it has the same key as prev_np of the `prevmap`, and if it is not, you are adding it. 
What I mean is that you don't need to loop, you can use `map::find` method which is much faster.